library/src/G4CMPPhononBoundaryProcess.cc

```diff
@@ -327,7 +328,15 @@ GetReflectedVector(const G4ThreeVector& waveVector,
 
     // FIXME: Find point on surface nearest to stepLocalPos, and reset
     surfAdjust = solid->DistanceToIn(stepLocalPos, -newNorm);	// Get distance along normal from new position back to detector surface
-    stepLocalPos -= surfAdjust * newNorm;			// Adjust position to be back on detector surface
+
+    // If surfAdjust > 1, we stepped off an edge and need to correct
+    if (surfAdjust > 1) {
+      stepLocalPos = GetEdgePosition(stepLocalPos, reflectedKDir, stepSize);
+      kTan = GetReflectionOnEdge(stepLocalPos, kTan);
+    } else {
```

Comment:
See my comments in the .hh file for updating value within function.  Take a look at `RotateToGlobalDirection()` for an example of such an _in situ_ update.